### PR TITLE
[Admin Portal]: Add resource ID param in the URL

### PR DESCRIPTION
**Pull request by devamanv**
### Which issue this PR addresses:
[ADO Card: SRE Portal v2 resourceid as a parameter](https://dev.azure.com/msazure/AzureRedHatOpenShift/_boards/board/t/Firefly/Stories/?workitem=15132772)

Fixes

### What this PR does / why we need it:
The PR contains changes  to the SRE portal so that it accepts Resource ID as a URI parameter which opens the detailed of the desired cluster directly. 

### Test plan for issue:

E2E tests have been added as part of the changes

### CODE REVIEW

File: portal/v2/src/ClusterList.tsx [hunk 1]
```diff
@@ -417,8 +491,11 @@ export function ClusterList(props: {
         styles={controlStyles}
       />
       <Separator styles={separatorStyle} />
-
+
```
Comment (ellis-johnson) [inline portal/v2/src/ClusterList.tsx:494]: Need to get rid of whitespace

**Review by ellis-johnson — COMMENTED** [review]
LGTM! Just one nitpick

File: portal/v2/src/ClusterList.tsx (hunk 1 above)
Comment (devamanv) [reply portal/v2/src/ClusterList.tsx]: Done

**Review by facchettos — DISMISSED** [review]
would be good to get rid of bindata here and use embed instead. edit: Nevermind, taken care of by shawn

File: portal/v2/src/ClusterList.tsx [hunk 2]
```diff
@@ -71,6 +76,59 @@ const separatorStyle = {
...
+                  window.history.replaceState({}, "", "/v2")
+                  props.hidePopup()
+                }}>
+                  Close
+                </DefaultButton>
+              </div>
+            </FocusTrapZone>
+          </Popup>
+        </Layer>
+    </>
```
Comment (jaitaiwan) [inline portal/v2/src/ClusterList.tsx:128]: There's only 1 top-level element `<Layer>` so the fragment `<>` is unnecessary.

File: portal/v2/src/index.tsx [hunk 3]
```diff
@@ -14,4 +14,19 @@ mergeStyles({
...
+  let paramStringFromURL = searchParams.split('?')[1];
+  let paramsArr = paramStringFromURL.split('&');
+
+  for (let i = 0; i < paramsArr.length; i++) {
+      let pair = paramsArr[i].split('=');
+      urlParams = {[pair[0]]: pair[1]}
+  }
+
+  return urlParams
+}
```
Comment (jaitaiwan) [inline portal/v2/src/index.tsx:30]: This is a helper function that should be extracted into some sort of libs file.

File: test/e2e/admin_portal.go [hunk 4]
```diff
@@ -244,4 +244,36 @@ var _ = Describe("Admin Portal E2E Testing", func() {
 			Expect(link.GetAttribute("href")).To(MatchRegexp(`https://([a-z]|[0-9])+\.admin\.aro\.azure\.com`))
 		}
 	})
+
+	It("Should open an error modal for an invalid resource ID parameter in the URL", func() {
+		wd.Get(host + "/v2" + "?resourceid=" + "invalidResourceId")
+		time.Sleep(time.Second * 2) // this is done to give enough time to fetch the data
+
+		wd.Wait(conditions.ElementIsLocated(selenium.ByCSSSelector, "div[role='document']"))
```
Comment (jaitaiwan) [inline test/e2e/admin_portal.go:252]: We shouldn't have both time.Sleep and wd.Wait. We should be waiting with specific enough selectors that it executes the test when the dom is ready with the data fetched.

File: test/e2e/admin_portal.go [hunk 5]
```diff
@@ -244,4 +244,36 @@ var _ = Describe("Admin Portal E2E Testing", func() {
...
+			SaveScreenshotAndExit(wd, err)
+		}
+
+		Expect(errorModal.IsDisplayed()).To(BeTrue())
+	})
+
+	It("Should display the correct cluster detail view for the resource ID parameter in the URL", func() {
+		wd.Get(host + "/v2" + "?resourceid=" + resourceIDFromEnv())
+		time.Sleep(time.Second * 2)
+		wd.Wait(conditions.ElementIsLocated(selenium.ByID, "ClusterDetailPanel"))
```
Comment (jaitaiwan) [inline test/e2e/admin_portal.go:264]: Same as previous comment

**Review by jaitaiwan — CHANGES_REQUESTED** [review]
Provided some more feedback on the PR. A few nicks a few best practice things.

**Comment by mociarain** [comment]
We have this now (unless I'm misunderstanding ResourceId in this context??)

Happy to close this?

**Comment by mociarain** [comment]
This has been implemented here AFAIK: https://github.com/Azure/ARO-RP/pull/2640.

Closing for now but do re-open if I've misunderstood

**Pull request closed without merging**